FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -4567,26 +4575,34 @@ subroutine GFS_physics_driver                                 &
             call set_state("ratio_of_snowfall_to_rainfall", Diag%sr)
             call set_state("tendency_of_rain_water_mixing_ratio_due_to_microphysics", rainp)
 
-            if (Model%emulate_zc_microphysics) then
-              ! apply microphysics emulator
-              call call_function("emulation", "microphysics")
-            endif
+            call call_function("emulation", "microphysics")
             
             if (Model%save_zc_microphysics) then
               call call_function("emulation", "store")
             endif
 
-            call get_state("air_temperature_output", Stateout%gt0)
+            call get_state("air_temperature_output", t_post_precpd)
             call get_state("specific_humidity_output", qv_post_precpd)
             call get_state("cloud_water_mixing_ratio_output", qc_post_precpd)
             call get_state("total_precipitation", rain1)
 
-            do k=1,levs
-              do i=1,im
-                Stateout%gq0(i,k,1) = qv_post_precpd(i,k)
-                Stateout%gq0(i,k,ntcw) = qc_post_precpd(i,k)
+            if (Model%ldiag3d) then
+              Diag%zhao_carr_emulator%humidity = (qv_post_precpd(1:im,1:levs) - dqdt(:,:,1)) / dtp
+              Diag%zhao_carr_emulator%cloud_water = (qc_post_precpd(1:im,1:levs) - dqdt(:,:,ntcw)) / dtp
+              Diag%zhao_carr_emulator%temperature = (t_post_precpd(1:im,1:levs) - dtdt) / dtp
+            end if
+
+            ! apply emulator
+            if (Model%emulate_zc_microphysics) then
+              do k=1,levs
+                do i=1,im
+                  Stateout%gq0(i,k,1) = qv_post_precpd(i,k)
+                  Stateout%gq0(i,k,ntcw) = qc_post_precpd(i,k)
+                  Stateout%gt0(i,k) = t_post_precpd(i,k)
```

Comment:
gt0 was not set before, while gq0 was, without any emulation conditional. isf emulate_zc_microphysics is False, is this ok?

Reply:
It's tricky, but i think this works. at this point gt0 contains the state last touched by the ZhaoCarr physics.

Previously gt0 was set in the `get_state` call above, but I made a new temporary array to be able to monitor the update before applying it
```
            call get_state("air_temperature_output", Stateout%gt0)
```